src/loki/reach.cc

```diff
@@ -86,21 +105,23 @@ directed_reach SimpleReach(const DirectedEdge* edge,
     return opp_edge->endnode();
   };
 
+  // NOTE: we can expand from the start of a complex or regular restriction because it cant possibly
+  // be on our path and we can expand from the end of a complex restriction because only the start
+  // would mark a potential stopping point (maybe a path followed the restriction)
```

Comment:
for the reverse direction its a bit easier because we always know to stop at the start of a complex or simple restriction. for the first edge though, we have no prior path so we can even start on the start of a simple or complex restriction.